fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java

```diff
@@ -91,13 +141,13 @@ public static ReferenceValue createReferenceValueFrom(Reference ref, String base
                         version = Integer.parseInt(tokens[3]);
                     }
                 }
+                // TODO: should the reference type be used if not null and we only have a logical ID?
```

Comment:
would be nice to resolve these TODOs if we can.  i'm not sure what the answer is but I think we have a better shot of getting it right now than in the future (since its all in our heads)

Reply:
I think it would make sense to use the type in the case where we only have a logical ID, but looking at the spec at  https://www.hl7.org/fhir/references.html#type, I see this line:

  `When the type is provided directly, it SHALL agree with the type determined by resolving the resource.`

Given that we don't attempt to resolve the reference at ingestion time, maybe this isn't a concern.

Reply:
I think we should handle it similar to how we handle absolute references.  Judging from the new lines just below, I guess we do use it.

But I remember @punktilious sharing his concern about potentially using these for compartment and/or chaining/includes. Hmmm...

How do we ensure that we don't attempt those for absolute references?  Maybe we should add a new value to the ReferenceType enum for this class of reference.  I *wouldn't* call it "Logical Id-only" but instead something like "non-FHIR relative" because the value wouldn't necessarilly need to match the resource's id...its basically just any relative reference that doesn't match the pattern for FHIR relative references.

Reply:
The existing code (prior to this change) does use the type for absolute references. I see no issue in treating logical ID-only references the same way. Basically it would then make the following two References equivalent:
```
 "subject": {
   "reference" : "Patient/123"
 }
```

```
 "subject": {
   "reference" : "123",
   "type" : "Patient"
 }
```

A logical ID-only reference without a type value would continue to be treated as it is today. In that case there's no concern with compartment/chain/include type searches, because the SQL generated for those searches always looks for a code_system_id of a specific resource type, so the logical ID-only reference would never match.

Reply:
yeah, my concern was that I don't know that we want a "logical id-only reference + a type" to work the same as an actual valid relative reference.

Reply:
team decision:  we don't want to handle these as the same;  in future our default will be to reject the "reference": "123" anyway.

Reply:
TODOs removed.